### DISCUSSION TITLE
Not connecting with Discord

### DISCUSSION THREAD

**Discussion by schmittyd** [fredemmott/StreamDeck-Discord#88] [Q&A]
**Describe the bug**
The integration is not always connecting with Discord. Clicking Reconnect or Reauthenticate does not force Discord to show the connection dialog.

**To Reproduce**

**Expected behavior**
Toggle Self-Mute and Toggle Self-Deafen should toggle respective actions in Discord

**Debugging Information**

![image](https://user-images.githubusercontent.com/12869594/126860195-1e9ed207-cb33-4f15-8365-d404446a4f0a.png)

**Log File**
https://gist.github.com/schmittyd/cec0d92474a29fd6e367ebedba791760

** Operating System **

Windows 10 Professional, 64-bit 19043.1149

**Additional context**

**Comment by Xercium** [comment]
I have the same issue, but for me it happens every time I reboot my pc. It loses the App ID and getting it to popup on discord again after copy pasting the app id back in is a pain. Sometimes it works, sometimes it does not.
From time to time it does not forget the app id and I can just press reconnect.
But I was using it for the longest time before without any issues.

What I changed before this started happening is that I have streamdeck running as admin now because it needs to be in admin for certain plugins and keypresses to work when running a game in admin, and I need to run the game in admin to be able to play it.
I also had to put my streamlabs in admin for it to keep working with my streamdeck.

I also have discord in admin now, but I think this might be causing issues, but not having it in admin and the buttons don't work anyway.

What a fine mess this is. Just to many different things needing to work together, I am close to giving up.
Also tried setting hotkeys for my discord mute and deafen, but they don't give feedback and also don't work in fullscreen game. :(

**Reply by Kobszx** [reply]
Fixed this problem by enabling run as administrator in application compatibility settings.

**Reply by Xercium** [reply]
You did this for what application?
And does it do it on startup of your pc or do you manually boot the applications?

**Reply by Kobszx** [reply]
I did it to stream deck. Enabling run as administrator in compatibility settings of the application removes it from automatically starting up on boot so you have to manually start stream deck every time you open the pc.

**Reply by Xercium** [reply]
Yes I am aware. I even got it to run as admin on startup and it works, but then I still need to boot discord manually as admin and open streamdeck to reconnect. This is for me not a solution to my issue. I don't want to be forced to manually boot or reboot programs to get the streamdeck to work as required.

**Comment by schmittyd** [comment]
This only started happening for me when the plugin was updated to v3, and Stream Deck was updated I think possibly three or four updates before v5.

**Comment by Xercium** [comment]
Did you try rolling back? I'm not even sure if it is possible.

**Comment by j3rseyhack** [comment]
Having the same issue, I have made 3 different applications, all with the https://localhost/ redirect, copied the id/security keys and it just keeps going to connecting and then connection_failed. logs just keep showing: PI Event: com.fredemmott.discord.rpc.reconnect and then PI Event: com.fredemmott.discord.rpc.reauthenticate over and over, never get the auth message.

**Comment by schmittyd** [comment]
I have figured out a workaround, although it is a PITA.

1. Quit Stream Deck app
2. Type `%AppData%\Elgato\StreamDeck\Plugins` in the File Explorer path bar
3. Find the `com.fredemmott.discord.sdPlugin` directory and delete it.
4. Start Stream Deck app
5. Install com.fredemmott.discord.streamDeckPlugin again.

I might have to figure out how to make some sort of script to delete the plugin on system shut down (after SD appp shuts down) and reinstall it on system start (before SD app starts). Unless @fredemmott can figure out how to fix the Discord auth problem.

**Reply by fredemmott** [reply]
If that fixes it, omething external (perhaps a false positive on anti-malware) is modifying the plugin after you've installed it. If you have the same problem again, can you please create a zip of the plugin directory and share it, before deleting it?

That might provide some information useful for debugging.

**Reply by Xercium** [reply]
For me I can reconnect every time if I first boot discord as admin and then open streamdeck and press reconnect. But I have to do this manually every time.

**Reply by schmittyd** [reply]
I've recently suffered a system crash and had to format and reinstall windows. It seems to be working fine now.

**Reply by Xercium** [reply]
I have to format at some point xD will update
